bundles/core/org.eclipse.smarthome.core.ephemeris/src/main/java/org/eclipse/smarthome/core/ephemeris/internal/EphemerisManagerImpl.java

```diff
@@ -0,0 +1,224 @@
+    private static final String CONFIG_DAYSET_WEEKEND = "weekend";
+    private static final String CONFIG_COUNTRY = "country";
+    private static final String CONFIG_REGION = "region";
+    private static final String CONFIG_CITY = "city";
+
+    private final Map<String, Set<DayOfWeek>> daysets = new HashMap<>();
+    private final Map<Object, HolidayManager> holidayManagers = new HashMap<>();
+    private final List<String> countryParameters = new ArrayList<>();
+
+    @NonNullByDefault({})
+    private String country;
+    @NonNullByDefault({})
+    private LocaleProvider localeProvider;
+
+    @Activate
+    protected void activate(Map<String, Object> config) {
+        modified(config);
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        config.entrySet().stream().filter(e -> e.getKey().startsWith(CONFIG_DAYSET_PREFIX)).forEach(e -> {
+            String[] setDefinition = e.getValue().toString().toUpperCase().split(",");
+            String[] setNameParts = e.getKey().split("-");
+            if (setDefinition.length > 0 && setNameParts.length > 1) {
+                Set<DayOfWeek> dayset = new HashSet<>();
+                Stream.of(setDefinition).forEach(day -> {
+                    dayset.add(DayOfWeek.valueOf(day));
+                });
+                daysets.put(setNameParts[1], dayset);
+            } else {
+                logger.warn("Erroneous dayset definition {} : {}", e.getKey(), e.getValue());
+            }
+        });
+
+        if (config.containsKey(CONFIG_COUNTRY)) {
+            country = config.get(CONFIG_COUNTRY).toString();
+        } else {
+            country = localeProvider.getLocale().getCountry();
+            logger.debug("Using system default country '{}' ", country);
+        }
+
+        if (config.containsKey(CONFIG_REGION)) {
+            countryParameters.add(config.get(CONFIG_REGION).toString());
+        }
+
+        if (config.containsKey(CONFIG_CITY)) {
+            countryParameters.add(config.get(CONFIG_CITY).toString());
+        }
+    }
+
+    @Reference
+    protected void setLocaleProvider(LocaleProvider localeProvider) {
+        this.localeProvider = localeProvider;
+    }
+
+    protected void unsetLocaleProvider(LocaleProvider localeProvider) {
+        this.localeProvider = null;
+    }
+
+    @Override
+    @Nullable
+    public Collection<ParameterOption> getParameterOptions(URI uri, String param, @Nullable Locale locale) {
+        List<ParameterOption> options = null;
+        if (CONFIG_URI.equals(uri.toString())) {
+            Locale nullSafeLocale = locale == null ? localeProvider.getLocale() : locale;
+            options = new ArrayList<>();
+            for (DayOfWeek day : DayOfWeek.values()) {
+                ParameterOption option = new ParameterOption(day.name(),
+                        day.getDisplayName(TextStyle.FULL, nullSafeLocale));
+                options.add(option);
```

Comment:
Something strange happens here - as this method creates new ParameterOption instances on every call, I see an ever increasing consolidated options list in the Paper UI upon every page refresh - all week days are always added additionally (instead of being merged with the existing ones). Do you see the same effect?

Reply:
Yes, I do reproduce the very same

Reply:
But I don't think they are merged at every page refresh. I think they collide with those defined in config.xml.

Reply:
=> Confirmed, it's a merge of values defined in the config.xml and getParameterOptions